### PR TITLE
Salesanalyst

### PR DESCRIPTION
**Pull request by beddings81**
Completed iteration 2. Still needs refactor!

### CODE REVIEW

File: lib/sales_analyst.rb [hunk 1]
```diff
@@ -64,4 +64,76 @@ def golden_items
...
+    top_days.map do |day|
+      day[0]
+    end
+  end
+
+  def invoice_status(status)
+    status_array = @engine.invoices.all.find_all do |invoice|
+      invoice.status == status
+    end
+  ((status_array.count / @engine.invoices.all.count.to_f) * 100).round(2)
```
Comment (Pocketzs) [inline lib/sales_analyst.rb]: The parenthesis on 137 needs to be pushed forward in line

File: lib/sales_analyst.rb [hunk 2]
```diff
@@ -64,4 +64,76 @@ def golden_items
...
+    (invoice_by_days.values.sum / invoice_by_days.values.length).round(2)
+  end
+
+  def average_invoices_per_day_standard_deviation
+    standard_deviation(invoice_by_days.values, invoice_average_per_day)
+  end
+
+  def top_days_by_invoice_count
+    std_dev = average_invoices_per_day_standard_deviation
+    top_days = invoice_by_days.find_all do |day, count|
```
[the remark this answers is not in the document]
Comment (Pocketzs) [reply lib/sales_analyst.rb:125]: Leave it alone

File: lib/sales_analyst.rb [hunk 3]
```diff
@@ -64,4 +64,76 @@ def golden_items
...
+      invoice.created_at.strftime('%A') 
+    end
+    day_hash = {
+      "Monday" => 0,
+      "Tuesday" => 0,
+      "Wednesday" => 0,
+      "Thursday" => 0,
+      "Friday" => 0,
+      "Saturday" => 0,
+      "Sunday" => 0
```
Comment (Pocketzs) [inline lib/sales_analyst.rb]: Use single quotes around the days

File: spec/sales_analyst_spec.rb [hunk 4]
```diff
@@ -186,9 +186,83 @@
...
+
+    expect(sales_analyst.top_days_by_invoice_count).to eq(["Monday"])
+  end
+
+  it 'can calculate percentages by the status' do
+    sales_engine = SalesEngine.from_csv(
+
+      :items     => './data/items_test.csv',
+      :merchants => './data/merchant_invoices_test2.csv',
+      :invoices  => './data/invoices_test2.csv'
```
[the remark this answers is not in the document]
Comment (Pocketzs) [reply spec/sales_analyst_spec.rb:260]: Ignore

File: spec/sales_analyst_spec.rb [hunk 5]
```diff
@@ -186,9 +186,83 @@
...
+    sales_analyst = sales_engine.analyst
+
+    expect(sales_analyst.top_days_by_invoice_count).to eq(["Monday"])
+  end
+
+  it 'can calculate percentages by the status' do
+    sales_engine = SalesEngine.from_csv(
+
+      :items     => './data/items_test.csv',
+      :merchants => './data/merchant_invoices_test2.csv',
```
[the remark this answers is not in the document]
Comment (Pocketzs) [reply spec/sales_analyst_spec.rb:259]: ignore

File: spec/sales_analyst_spec.rb [hunk 6]
```diff
@@ -186,9 +186,83 @@
...
+    )
+    sales_analyst = sales_engine.analyst
+
+    expect(sales_analyst.top_days_by_invoice_count).to eq(["Monday"])
+  end
+
+  it 'can calculate percentages by the status' do
+    sales_engine = SalesEngine.from_csv(
+
+      :items     => './data/items_test.csv',
```
[the remark this answers is not in the document]
Comment (Pocketzs) [reply spec/sales_analyst_spec.rb:258]: ignore

File: spec/sales_analyst_spec.rb [hunk 7]
```diff
@@ -186,9 +186,83 @@
...
+      :invoices  => './data/invoices_test2.csv'
+    )
+    sales_analyst = sales_engine.analyst
+
+    expect(sales_analyst.top_days_by_invoice_count).to eq(["Monday"])
+  end
+
+  it 'can calculate percentages by the status' do
+    sales_engine = SalesEngine.from_csv(
+
```
Comment (Pocketzs) [inline spec/sales_analyst_spec.rb]: probably delete line 257

File: spec/sales_analyst_spec.rb [hunk 8]
```diff
@@ -186,9 +186,83 @@
...
+  it 'can return the top days of the week by invoice count' do
+    sales_engine = SalesEngine.from_csv(
+
+      :items     => './data/items_test.csv',
+      :merchants => './data/merchant_invoices_test2.csv',
+      :invoices  => './data/invoices_test2.csv'
+    )
+    sales_analyst = sales_engine.analyst
+
+    expect(sales_analyst.top_days_by_invoice_count).to eq(["Monday"])
```
[the remark this answers is not in the document]
Comment (Pocketzs) [reply spec/sales_analyst_spec.rb:252]: single quote monday

**Review by Pocketzs — CHANGES_REQUESTED** [review]
The following comments contain what I think should be changed or ignored

**Review by KaraJoHo — APPROVED** [review]

**Review by Kledin85 — APPROVED** [review]
looks good.

**Review by Pocketzs — APPROVED** [review]

**Pull request merged**